### ISSUE TITLE
Support hot reload

### ISSUE THREAD

**Issue by elliotBraem** [NEARBuilders/near-bos-webcomponent#20]
In order to be able to fully replace the bos-workspace default gateway with near-bos-webcomponent, we need to introduce hot reloading.

Pull over the useRedirectHook from [bos-workspace/gateway](https://github.com/NEARBuilders/bos-workspace/blob/main/gateway/src/useRedirectMap.js). This can also do some code clean-up in App.js -> Viewer, although maintain the "nearSocialVMredirectMap" SESSION_STORAGE_REDIRECT_MAP_KEY

Include documentation on how to use redirect map with hot reload (web socket) or session storage

**Comment by petersalomonsen** [comment]
Remember that "Don't ask again"  ( https://github.com/NEARBuilders/bos-workspace/issues/114 ) does not work with the redirect map. I believe that rather than using a redirectmap, a local RPC proxy should be provided that intercepts the calls that the VM does to get widget content, replacing the result with local widget content.

**Comment by elliotBraem** [comment]
I just investigated #114 and found the issue: it was misuse of the "commander" package for the cli. I've opened a [pull request here](https://github.com/NEARBuilders/bos-workspace/pull/115) to fix it. With this change, I have confirmed that disabling hot reload (`bw dev --no-hot`) will not use redirectMap in bos-workspace gateway, and will only use the rpc proxy that bos-workspace provides, see below:

![image](https://github.com/NEARBuilders/near-bos-webcomponent/assets/16282460/e5a14095-08b8-4d42-8324-e986ca4be8b5)

![image](https://github.com/NEARBuilders/near-bos-webcomponent/assets/16282460/37ad3941-effe-44ea-9829-88991648f217)

---

Now regarding redirectmap, maybe we should discuss more for clarity:

In the pull request #26, the "useRedirectMap" hook handles three scenarios:

 1. local storage "flags" holding bosLoaderUrl (backwards compatibility with [bos-loader](https://github.com/near/bos-loader))
 2. session storage redirect map key (support devhub test solution)
 3. hot reload enabled by "enablehotreload" attribute on the web component, gets url from environment or hardcoded (meh)

I agree that rpc proxy is much better and replaces the need for 1 + 2 -- and if we had to keep supporting one of these two, I would only keep session storage, because using bosLoaderUrl can be unpredictable (e.g. what port is it running on?). Although if we don't need to support either, that'd be nice too!

My hope is to replace bos-workspace gateway with near-bos-webcomponent, and adding support for hot reload and ether.js are the only two items missing. And as shown in the fix, I think with hotreloadenabled = false, we can circumvent the redirectMap altogether and it will only use RPC proxy, satisfying the need when running a gateway for tests.
